sound/soc/sof/pcm.c

```diff
@@ -113,6 +134,12 @@ static int sof_pcm_hw_params(struct snd_soc_component *component,
 	if (!spcm)
 		return -EINVAL;
 
+	if (spcm->prepared[substream->stream]) {
```

Comment:
It's probably worth a comment here explaining the reason for this check and subsequent action.

Reply:
@lgirdwood Ack, added the comment now.